app/src/ui/branches/ci-check-run-list.tsx

```diff
@@ -106,17 +108,42 @@ export class CICheckRunList extends React.PureComponent<
   private onStatus = async (check: ICombinedRefCheck | null) => {
     const statusChecks = check !== null ? check.checks : []
 
-    const checkRuns =
-      statusChecks.length > 0
-        ? await this.props.dispatcher.getActionsWorkflowRunLogs(
-            this.props.repository,
-            this.getCommitRef(this.props.prNumber),
-            this.props.branchName,
-            statusChecks
-          )
-        : statusChecks
-
-    this.setState({ checkRuns, loadingLogs: false })
+    if (statusChecks.length === 0) {
+      this.setState({
+        checkRuns: statusChecks,
+        loadingActionLogs: false,
+        loadingActionWorkflows: false,
+      })
+      return
+    }
+
+    /*
+      Until we retrieve the actions workflows, we don't know if a check run has
+      action logs to output, thus, we want to show loading until then. However,
+      once the workflows have been retrieved and since the logs retrieval and
+      parsing can be noticeably time consuming. We go ahead and flip a flag so
+      that we know we can go ahead and display the checkrun `output` content if
+      a check run does not have action logs to retrieve/parse.
+    */
+    const checkRunsWithActionsUrls = await this.props.dispatcher.getCheckRunActionsJobsAndLogURLS(
+      this.props.repository,
+      this.getCommitRef(this.props.prNumber),
+      this.props.branchName,
+      statusChecks
+    )
+
+    this.setState({
```

Comment:
Given that these methods can end up taking a bit of time it's pretty easy to get into a scenario where we call setState after unmount

![image](https://user-images.githubusercontent.com/634063/136947533-dd9b45ec-2c4c-49a7-b4d1-dca15a3d4f68.png)

Could we either add a flag that we toggle on unmount (i.e. the [good ol' so-called anti-pattern](https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html#:~:text=An%20easy%20migration%20strategy%20for%20anyone%20upgrading%20their%20code%20to%20avoid%20isMounted()%20is%20to%20track%20the%20mounted%20status%20yourself.%20Just%20set%20a%20_isMounted%20property%20to%20true%20in%20componentDidMount%20and%20set%20it%20to%20false%20in%20componentWillUnmount%2C%20and%20use%20this%20variable%20to%20check%20your%20component%E2%80%99s%20status.)) or could we use `this.subscription` as a proxy for that?

```suggestion
  if (this.subscription === null) {
    return
  }

  this.setState({
```

If nothing else it'll prevent us from calling `getActionsWorkflowRunLogs` unnecessarily